### ISSUE TITLE
Enable/disable tiling per workspace

### ISSUE THREAD

**Issue by diegobernardes** [pop-os/shell#380]
**Issue/Bug Description:**
It would be awesome to enable the tiling per workspace. As a developer, I may be working in a workspace that has my editor and some information, but on other workspaces, I may want to have free-floating windows in order to do my job, maybe browsers and other GUI apps.

**Expected behavior:**
Be able to enable/disable the tiling per workspace.

**Comment by bitmux** [comment]
Per workspace and possibly an exclusion option per application.  Some applications simply cannot handle being tiled.  

Expected behavior: right-click menu on application title bar should include an option to exclude this application from being auto-tiled

**Comment by weebs** [comment]
> Per workspace and possibly an exclusion option per application. Some applications simply cannot handle being tiled.
> 
> Expected behavior: right-click menu on application title bar should include an option to exclude this application from being auto-tiled

The exclusion per application would be very helpful, and maybe a configurable rule for auto-exclusion. In particular I've had issues with borderless windowed games getting tiled. It might make sense to have an option to automatically float windows that match or exceed the monitor resolution

**Comment by rodelrod** [comment]
For some use cases I need partially overlapping windows in one workspace. In a nother workspace, I may have floating windows arranged in some way. 

When I turn on tiling in one workspace, I do not want to break the arrangement in workspaces that I cannot see or maybe even remember.

Enabling the tiling per workspace would be IMHO the deadliest of killer features for widespread adoption of tiling and Pop Shell is in the sweet spot to provide it! 

Thanks for your great work and here's hoping that this feature request gets some traction.

**Comment by VictorMeneghini** [comment]
is there any update about it? (:

**Comment by stuarthoughton** [comment]
Agreed. This would be really useful for me. I generally prefer floating windows but having a browser auto-tile with a text editor so I can copy data from one to the other is really useful. Shoving that functionality onto a single workspace would be ideal.

**Comment by jakeperry** [comment]
It would be cool to enable tiling per monitor also.

**Comment by shirazawan** [comment]
Still nothing :(

**Comment by vtevelde** [comment]
Yes please!! Per-workspace tiling is an essential missing feature. Switching from Xmonad in order to have gnome compatibility was mostly seamless except for this issue.

**Comment by VaZark** [comment]
This is particularly useful when merged with separate workspaces per monitor. I usually have docs or a browser open on the secondary display while the primary has the IDE, terminal and slack on separate workspaces.

Being able to tile only my IDE and terminal workspaces would be amazing.

**Comment by maria-komarova** [comment]
Hi all, we're currently developing per workspace tiling feature for COSMIC desktop environment. If anyone is interested in participating in the short usability testing of the feature in the design program, please reach out to me at ux@system76.com this week.

**Comment by bitmux** [comment]
I'd be interested in  assisting with usability testing for the tiling
feature!  Thanks for the invitation.

-----Original Message-----
From: maria-komarova ***@***.***>
Reply-To: pop-os/shell
***@***.***>
To: pop-os/shell ***@***.***>
Cc: bitmux ***@***.***>, Comment
***@***.***>
Subject: Re: [pop-os/shell] Enable/disable tiling per workspace (#380)
Date: Tue, 26 Sep 2023 11:39:02 -0700

Hi all, we're currently developing per workspace tiling feature for
COSMIC desktop environment. If anyone is interested in participating in
the short usability testing of the feature in the design program,
please reach out to me at ***@***.*** this week.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you commented.Message ID: <pop-
***@***.***>

**Comment by maria-komarova** [comment]
Thanks for your interest in testing @bitmux. I'm doing the testing via online calls. If you could provide some way to reach you or just send an email to ux@system76.com I can forward you more information and figure out the scheduling.

**Comment by felipebueno** [comment]
Hello, @maria-komarova!

Are you still testing the "per workspace tiling" feature? I'm an i3wm user trying to adapt the COSMIC desktop to my workflow so I'm really interested in this feature.

**Comment by leviport** [comment]
@felipebueno This feature likely will not be incorporated into the current Gnome-based COSMIC desktop, but it does currently exist in our new DE, [COSMIC DE](https://github.com/pop-os/cosmic-epoch). It should be noted that COSMIC DE is still pre-alpha and many things are still a work in progress, but it is currently quite usable, if you want to check it out.

**Comment by maria-komarova** [comment]
@felipebueno I've finished the usability testing of the designs some time ago but I can let you know if we need to do any more usability testing. As was mentioned, the functionality is currently present in COSMIC DE although the GUI still needs implementation, or at least a part of it.
